### DISCUSSION TITLE
No wheels on pypi for MacOS < 10.15 since 0.5

### DISCUSSION THREAD

**Discussion by hoesler** [alan-turing-institute/sktime#723] [Q&A]
Any specific reason you dropped support for older MacOS versions?
https://pypi.org/project/sktime/0.5.3/#files

**Comment by fkiraly** [comment]
Moved to discussion since a question.
We can raise an issue if this turns into a work item.

I believe @mloning was the last to update CI/deployment workflows, some relevant issues/PR:

https://github.com/alan-turing-institute/sktime/issues/196
https://github.com/alan-turing-institute/sktime/pull/505

@mloning may be slow to respond though, since working on his thesis.
A plausible reason might be that this specific set-up wasn't covered by sufficient support by the community.

Generally, we rely heavily on volunteer contributions, so proposed actions or solutions are very much welcome :smiley:

**Reply by hoesler** [reply]
Well, I don't think that I am really able to contribute to your build and publishing process without access to the CI/CD environment. I might be wrong, but I think all you need to do is to [set the mac os x runner](https://github.com/alan-turing-institute/sktime/blob/997b318ba677fc14e4f4a08c2f2a1664ab56a451/.github/workflows/upload-to-pypi.yml#L12) to an earlier OS version.

If you want to discuss, if that is really an issue or not: From my perspective, I see no point in excluding users of a still actively maintained OS version in using your package from pypi, as it can be installed from source without a problem.

**Comment by mloning** [comment]
@hoesler thanks for reaching out, I'm not sure if we can easily support an older version. As far as I understand GitHub only supports macOS 10 or newer, see the documentation on supported macOS runners [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners). Any way to still support older versions, perhaps on a different CI service?

**Reply by ghost** [reply]
Just to make sure, I am not talking about versions older than 10, but older than 10.15, especially about 10.14.
You're right however, that macOS runners older than Catalina don't seem to be supported.
But you can set a `MACOSX_DEPLOYMENT_TARGET` env var for python:
https://github.com/actions/virtual-environments/issues/1256#issuecomment-770270252

**Reply by mloning** [reply]
So we could iterate over those versions in the workflow [here](https://github.com/alan-turing-institute/sktime/blob/master/.github/workflows/upload-to-pypi.yml). 

Would 10.14 be sufficient or would you suggest to go back even further?

**Reply by ghost** [reply]
I would at least support OS versions which haven't reached end of life yet. So, I think this would be 10.14.
But if you look at [one of the the PRs](https://github.com/PyTables/PyTables/pull/872/files) linked in the issue I posted, you'll find this:
```
          # Best compatibility, even with older releases of macOS.
          MACOSX_DEPLOYMENT_TARGET: "10.9"
```

**Reply by mloning** [reply]
I'd appreciate a PR for 10.14 as a start, we will have to see if all our dependencies support that version and if all our tests pass.

**Reply by ghost** [reply]
I will have a look. I would favour 10.13 though, to be consistent with the 0.4 releases.
